### PR TITLE
[docs] Add info callout about available component `slots`

### LINKED ISSUE #7683: [docs] Teach API slots documentation
The page https://mui.com/x/react-date-pickers/custom-components/ could have a section to explain that API pages list elements such as: https://mui.com/x/api/date-pickers/date-picker/#slots

From slack message

>	Please declare all the components that are customizable.
	sent from https://mui.com/x/react-date-pickers/custom-components/ (from section Component)

### PR DESCRIPTION
**Pull request by Vivek-Prajapatii**
Fix #7683

information about the customizable components is added

### CODE REVIEW

File: docs/data/date-pickers/custom-components/custom-components.md [hunk 1]
```diff
@@ -6,7 +6,12 @@ components: DateTimePickerTabs
 
 # Custom components
 
-<p class="description">The date picker lets you customize sub-components.</p>
+<p class="description">The date picker lets you customize sub-components.
+
+:::info
+The components which can be customized are listed under `slots` section in DatePicker API Reference.
+:::
+</p>
```
Comment (LukasTy) [inline docs/data/date-pickers/custom-components/custom-components.md:14]: What do you think about the following suggestion. It's adding links and increasing the description specificity.
Also fixing the `p` element ending location.

```suggestion
<p class="description">The date picker lets you customize sub-components.</p>

:::info
The components that can be customized are listed under `slots` section in Date and Time Pickers [API Reference](/x/api/date-pickers/).
For example, available Date Picker slots can be found [here](/x/api/date-pickers/date-picker/#slots).
:::
```

**Review by LukasTy — COMMENTED** [review]
Thank you for contribution and a good suggestion! 💯 
What do you think about my suggested improvement? 🤔

**Comment by alexfauquette** [comment]
Just linked your PR to the issue you're fixing (#7683) such that it get closed when the PR is merged.


If you want more information about [how to link PR with issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) you can check the GitHub documentation :)

File: docs/data/date-pickers/custom-components/custom-components.md (hunk 1 above)
Comment (Vivek-Prajapatii) [reply docs/data/date-pickers/custom-components/custom-components.md:14]: sure @LukasTy ill update the PR, Thanks you for the suggestion

**Comment by Vivek-Prajapatii** [comment]
> Just linked your PR to the issue you're fixing (#7683) such that it get closed when the PR is merged.
> 
> If you want more information about [how to link PR with issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) you can check the GitHub documentation :)

Thank you for linking the issue @alexfauquette , i think i don't have the permissions to do the same

**Review by LukasTy — APPROVED** [review]
Thank you for your contribution!

FYI: For some reason the pipeline "did not want" to run for you, so I intervened with my commits and also fixed the prettier/linting issue. 😉

**Comment by Vivek-Prajapatii** [comment]
> Thank you for your contribution!
> 
> FYI: For some reason the pipeline "did not want" to run for you, so I intervened with my commits and also fixed the prettier/linting issue. wink

thanks @LukasTy

**Pull request merged**